frontend/src/components/data-table/column-pinning/feature.ts

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+import {
+  Column,
+  ColumnPinningOptions,
+  ColumnPinningPosition,
+  ColumnPinningState,
+  ColumnPinningTableState,
+  makeStateUpdater,
+  RowData,
+  Table,
+  TableFeature,
+  Updater,
+} from "@tanstack/react-table";
+
+export const ColumnPinningFeature: TableFeature = {
```

Comment:
i think this is already built into `tanstack`. do you think it would be possible to re-use that?

1. either `__select__` is always pinned (setting it in the initial state)
2. keeping this table feature, but overriding the existing functions if necessary. not sure if its possible, but something like:
```ts
const originaPin = column.pin;
if (originaPin) {
   column.pin = () => {
      // do some custom things
       return originaPin();
   }
}
```


if its not possible, then this is ok, but ideally we can use the internals as much as possible

Reply:
You know, now that I'm looking at this, I'm not really sure why I implemented it myself. Lol

Reply:
removed the "custom" feature -- `ColumnPinning` from tanstack works like a charm